File: contracts/wasm/helloworld/ts/helloworld/helloworld.ts
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import * as wasmlib from "wasmlib"
import * as sc from "./index";

export function funcHelloWorld(ctx: wasmlib.ScFuncContext, f: sc.HelloWorldContext): void {
    ctx.log("Hello, world!");
}

export function viewGetHelloWorld(ctx: wasmlib.ScViewContext, f: sc.GetHelloWorldContext): void {
    f.results.helloWorld().setValue("Hello, world!");
}

File: contracts/wasm/helloworld/go/helloworld/helloworld.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package helloworld

import (
	"github.com/iotaledger/wasp/packages/wasmvm/wasmlib/go/wasmlib"
)

//nolint:unparam
func funcHelloWorld(ctx wasmlib.ScFuncContext, f *HelloWorldContext) {
	ctx.Log("Hello, world!")
}

func viewGetHelloWorld(ctx wasmlib.ScViewContext, f *GetHelloWorldContext) {
	f.Results.HelloWorld().SetValue("Hello, world!")
}
